### PR TITLE
[99251] Enhance Resumable Upload Failure Logging

### PR DESCRIPTION
**Pull request by OnnaRob**
When we upload files to Google and a failure is hit, capture the `Content-Range` header to determine if there is an issue with it. Based on the docs from Google, it is likely there is an issue with what we are sending, but we have no way of knowing that without this output.

### CODE REVIEW

**Comment by OnnaRob** [comment]
https://app.shortcut.com/onnahq/story/99251/guillotina-gcs-library-occasionally-errors-when-uploading

File: guillotina_gcloudstorage/storage.py [hunk 1]
```diff
@@ -264,7 +264,10 @@ async def _append(self, dm, data, offset):
                             "info": text,
                         }
                     )
-                raise GoogleCloudException(f"{call.status}: {text}")
+                content_range = headers["Content-Range"]
```
Comment (gitcarbs) [inline guillotina_gcloudstorage/storage.py]: Will this always be available, should we be more defensive with a `.get("Content-Range", "")`?

Comment (OnnaRob) [reply guillotina_gcloudstorage/storage.py]: Only available when we're actually chunking an upload, good call.

**Review by gitcarbs — APPROVED** [review]

**Review by amorillas — APPROVED** [review]

**Comment by OnnaRob** [comment]
https://app.shortcut.com/onnahq/story/99251/guillotina-gcs-library-occasionally-errors-when-uploading

**Pull request merged**